osu.Game.Rulesets.Taiko/Skinning/Legacy/LegacyCirclePiece.cs

```diff
@@ -115,14 +120,8 @@ private void animateForegroundLayer(IFramedAnimation animatableForegroundLayer)
                 return;
             }
 
-            if (beatSyncProvider?.ControlPoints != null)
-            {
-                beatLength = beatSyncProvider.ControlPoints.TimingPointAt(Time.Current).BeatLength;
-
-                animationFrame = Time.Current % ((beatLength * 2) / multiplier) >= beatLength / multiplier ? 0 : 1;
-
-                animatableForegroundLayer.GotoFrame(animationFrame);
-            }
+            animationFrame = Math.Abs(Time.Current - timingPoint.Time) % ((timingPoint.BeatLength * 2) / multiplier) >= timingPoint.BeatLength / multiplier ? 0 : 1;
```

Comment:
The change with `timingPoint.Time` here makes sense, but the optimisation of fetching the control point once means that potentially different hitcircles will animate with a different frequency in case of a bpm change.

It's a bit hard to describe so I'll try with video.

current master:

https://github.com/ppy/osu/assets/20418176/6ce47888-6c3b-47d5-a3ca-a0abcf715804

this PR:

https://github.com/ppy/osu/assets/20418176/2ce73ef5-f3e8-4b36-a431-5b3d4205a536

freeze frames side by side:

| master | this PR |
| :-: | :-: |
| ![1688593156](https://github.com/ppy/osu/assets/20418176/0283ccf9-eda9-45ef-96cb-93135890da82) | ![1688593185](https://github.com/ppy/osu/assets/20418176/85c4be7c-8b68-422e-9854-bd90adef281a) |

Beatmap used to test: [test - test3 (spaceman_atlas2_old).osz.zip](https://github.com/ppy/osu/files/11962300/test.-.test3.spaceman_atlas2_old.osz.zip) (delete last `.zip` extension, was only added so that github accepts the file).

Not sure how much of a deal this is, but bringing it up anyways.

Reply:
I don't particularly mind this myself, but it's not what stable does, as far as I can tell.

If you're fine with changing this then so am I, I mostly just want to avoid a scenario wherein we have to go back to this to fix it again because someone says it's not like stable. I'll leave an approval here to that end, if you're fine with breaking behaviour, then merge at your leisure.

Reply:
There's a 50% chance people are going to ask for it to be changed, but I'd like to wage against the odds here because I think the new behaviour makes more sense.

Sorry in advance if we do need to revisit this, I guess 🤷